beep/tests/test_structure.py

```diff
@@ -197,7 +197,7 @@ def test_to_diagnostic_interpolation_cycler_run(self):
 
         v_range, resolution, nominal_capacity, full_fast_charge, diagnostic_available = \
             cycler_run.determine_structuring_parameters()
-        diag_interpolated = cycler_run.diagnostic_interpolated(diagnostic_available)
+        diag_interpolated = cycler_run.get_interpolated_diagnostic_cycles(diagnostic_available)
         diag_cycle = diag_interpolated[(diag_interpolated.cycle_type == 'rpt_0.2C')
```

Comment:
Lmk if you plan to, or if you would me like to add an assert for the additional grouping by `step_index_counter` that I included in a recent PR.

Reply:
There's a revised test in `test_get_interpolated_diagnostic_cycles` that asserts that the partitioned steps have the same length, which roughly reflects their separate treatment, but I think ultimately we'll need to shore up the functionality more generally to have a more robust test.